fix: Fixed loading changelog lint config

rpl opened the pull request:
Fixes #767

This PR introduces a small change in the `travis-pr-title-lint` grunt task that fixes the `conventional-changelog-lint` config loading.

Based on the failure from this Travis build https://travis-ci.org/mozilla/web-ext/jobs/193576176, we noticed that the configuration file `'.conventional-changelog-lintrc'` was not loading correctly, by looking at the `conventional-changelog-lint`, sources it looks like the reason was related to the usage of `getConfiguration` with the wrong parameters (which unfortunately doesn't generate any complains or promise rejection, it just return a default configuration based on the selected preset).

rpl commented:
@kumar303 This PR should fix the regression described by #767

kumar303 commented on tasks/travis-pr-title-lint.js:
```diff
@@ -154,7 +154,7 @@ module.exports = function(grunt) {
 
       Promise.all([
         getPreset('angular'),
-        getConfiguration('.conventional-changelog-lintrc'),
```
Oh man, I can't believe they don't raise an error for this. How about adding some annoying code to check the file since they don't raise an error? Something like:
````javascript
var config = path.join(__dirname, '..', '.conventional-changelog-lintrc');
var configStat = file.statSync(config);
if (!configStat || !configStat.isFile()) {
  throw new Error(...);
}
````

rpl replied:
:+1:

rpl commented:
@kumar303 I've just updated this PR with the proposed tweaks (I opted to use 'mz/fs' which provides the promisified version of the fs module, and I added both an fs.stat and an fs.access check, and so it is going to check both that the file exists, that is readable and that is a regular file).

kumar303 commented on tasks/travis-pr-title-lint.js:
```diff
@@ -152,12 +154,30 @@ module.exports = function(grunt) {
+        checkLintConfig,
         getPreset('angular'),
-        getConfiguration('.conventional-changelog-lintrc'),
+        getConfiguration(),
```
hmm, why do you no longer need to pass in the file? Is it because we can assume it will be auto-detected? I thought it would be: `getConfiguration(config)`

rpl replied:
yep, unfortunately the getConfiguration helper doesn't take the filename, but a series of parameter that will affect the final config file name :-)

Code References:
- https://github.com/marionebl/conventional-changelog-lint/blob/875a77d66cf09f8106fc348e85b8a6ddcbcfa55d/source/cli.js#L103-L109
- https://github.com/marionebl/conventional-changelog-lint/blob/master/source/library/get-configuration.js#L11-L17

kumar303 replied:
wow, that's confusing!
